samlauthenticator/samlauthenticator.py

```diff
@@ -724,9 +726,27 @@ def _get_redirect_from_metadata_and_redirect(authenticator_self, element_name, h
 
         redirect_link_getter = xpath_with_namespaces(final_xpath)
 
+        return redirect_link_getter(saml_metadata_etree)[0]
+
+    def _get_redirect_from_metadata_and_redirect(authenticator_self, element_name, handler_self, add_authn_request=False):
+
+        redirect_url = authenticator_self._get_redirect_from_metadata(element_name, handler_self) 
+
         # Here permanent MUST BE False - otherwise the /hub/logout GET will not be fired
         # by the user's browser.
-        handler_self.redirect(redirect_link_getter(saml_metadata_etree)[0], permanent=False)
+        if add_authn_request:
+            authn_requst = quote_plus(b64encode(zlib.compress(
+                authenticator_self._make_authn_request(element_name, handler_self).encode('utf8')
+                )[2:-4]))
+            handler_self.redirect(
+                f"{redirect_url}?SAMLRequest={authn_requst}", 
```

Comment:
if redurect url already has queryparametar, question mark will be appeared twice.

actually google workspace saml include idpid parameter in redirect URL. if that case redirection will fail.